File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/06_Nested_Loops_Exercise/05_Special_Numbers.js
function specialNumbers(input) {
    let num = Number(input[0]);

    let result = '';

    for (let currentNum = 1111; currentNum <= 9999; currentNum++) {
        let currentNumAsString = String(currentNum);
        let isValid = true;

        for (let i = 0; i < currentNumAsString.length; i++) {
            let currentDigitAsText = currentNumAsString[i];
            let currentDigit = Number(currentDigitAsText);

            if (num % currentDigit != 0) {
                isValid = false;
                break;
            }
        }

        if (isValid) {
            result += `${currentNumAsString} `;
        }
    }

    console.log(result);
}


// specialNumbers(["3"]);

File: Python_Courses/Python_Basics/Python_PB_2025_03/06_Nested_Loops_Exercise/05_Special_Numbers.py
number = int(input())

# 1. взимаме всяко число от 1_111 до 9_999
for num in range(1_111, 10_000):
    # 2. проверяваме дали числото е 'специално'

    is_special = True
    num_as_str = str(num)  # str(2418) == '2418'
    for d in num_as_str:  # ['2', '4', '1', '8']
        d_as_int = int(d)  # int('2') == 2

        if d_as_int == 0:
            is_special = False
            break

        if number % d_as_int != 0:
            is_special = False
            break
    #    -> ако е специално => печатим го на конзолата
    if is_special:
        print(num, end=' ')
